athena-gcs/src/test/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandlerTest.java

```diff
@@ -318,6 +316,46 @@ public void testDoGetSplits() throws Exception
         );
         when(table.getPartitionKeys()).thenReturn(columns);
         GetSplitsResponse response = gcsMetadataHandler.doGetSplits(blockAllocator, request);
-        assertNotNull(response);
+        assertEquals(2, response.getSplits().size());
+        assertEquals(2, response.getSplits().stream().map(split -> split.getProperties().get("year")).count());
+    }
+
+    @Test
+    public void testDoGetSplitsProperty() throws Exception
+    {
+        Schema schema = SchemaBuilder.newBuilder()
+                .addStringField("yearCol")
+                .addStringField("monthCol")
+                .build();
+        BlockAllocatorImpl allocator = new BlockAllocatorImpl();
+        Block partitions = allocator.createBlock(schema);
+
+        int num_partitions = 10;
+        for (int i = 0; i < num_partitions; i++) {
+            BlockUtils.setValue(partitions.getFieldVector("yearCol"), i, 2016 + i);
+            BlockUtils.setValue(partitions.getFieldVector("monthCol"), i, (i % 12) + 1);
+        }
+        partitions.setRowCount(num_partitions);
+        GetSplitsRequest request = new GetSplitsRequest(federatedIdentity,
+                QUERY_ID, CATALOG, TABLE_NAME,
+                partitions, com.google.common.collect.ImmutableList.of("yearCol", "monthCol"), new Constraints(new HashMap<>()), null);
+        QueryStatusChecker queryStatusChecker = mock(QueryStatusChecker.class);
+        when(queryStatusChecker.isQueryRunning()).thenReturn(true);
+        GetTableResult getTableResult = mock(GetTableResult.class);
+        StorageDescriptor storageDescriptor = mock(StorageDescriptor.class);
+        when(storageDescriptor.getLocation()).thenReturn(LOCATION);
+        Table table = mock(Table.class);
+        when(table.getStorageDescriptor()).thenReturn(storageDescriptor);
+        when(table.getParameters()).thenReturn(com.google.common.collect.ImmutableMap.of(PARTITION_PATTERN_KEY, "year=${yearCol}/month${monthCol}/", CLASSIFICATION_GLUE_TABLE_PARAM, PARQUET));
+        when(awsGlue.getTable(any())).thenReturn(getTableResult);
+        when(getTableResult.getTable()).thenReturn(table);
+        List<Column> columns = com.google.common.collect.ImmutableList.of(
+                createColumn("yearCol", "varchar"),
+                createColumn("monthCol", "varchar")
+        );
+        when(table.getPartitionKeys()).thenReturn(columns);
+        GetSplitsResponse response = gcsMetadataHandler.doGetSplits(blockAllocator, request);
+        assertEquals(10, response.getSplits().size());
+        assertEquals(10, response.getSplits().stream().map(split -> split.getProperties().get("year")).count());
```

Comment:
Same here, you are only still checking for the count here.
You need to validate that the values are what we expect.